docs/book/misc/handling-dependencies.md

```diff
@@ -0,0 +1,42 @@
+---
+description: How to handle issues with conflicting dependencies
+---
+
+# Dependency Resolution and ZenML
+
+This page documents a some of the common issues that arise when using ZenML with other libraries.
+
+**Last Updated**: February 20, 2023
+
+When using ZenML with other libraries, you may encounter issues with conflicting
+dependencies. ZenML aims to be stack- and integration-agnostic, allowing you to
+run your pipelines using the tools that make sense for your problems. With this
+flexibility comes the possibility of dependency conflicts.
+
+## Suggestions for Resolving Dependency Conflicts
+
+### Use a tool like `pip-compile` for reproduciblity
+
+Consider using a tool like `pip-compile` (available through [the `pip-tools`
+package](https://pip-tools.readthedocs.io/)) to compile your dependencies into a
+static `requirements.txt` file that can be used across environments.
+
+### Well-known dependency resolution issues
```

Comment:
You should add another section that explains how they can use `pip check` to show possible dependency conflicts. Same idea here: they might be harmless, but users should be aware of them in case problems do occur when they run pipelines.

Reply:
Here's one more idea: mention what happens if the dependencies required by zenml integrations are not respected: some components might stop working or throwing errors. A good way to check if these are met is to use `zenml integration list` to see if an integration is active.